plugins/azure_sentinel/plugin.spec.yaml

```diff
@@ -0,0 +1,879 @@
+    lastActivityTimeUtc:
+      description: The time of the last activity in the incident
+      title: The time of the last activity in the incident
+      type: date
+    lastModifiedTimeUtc:
+      description: The last time the incident was updated
+      title: The last time the incident was updated
+      type: date
+    name:
+      description: Azure resource name
+      title: Azure resource name
+      type: string
+    owner:
+      description: Describes a user that the incident is assigned to
+      title: Describes a user that the incident is assigned to
+      type: IncidentOwnerInfo
+    relatedAnalyticRuleIds:
+      description: List of resource ids of Analytic rules related to the incident
+      title: List of resource ids of Analytic rules related to the incident
+      type: "[]string"
+    severity:
+      title: Incidents severity
+      description: Incidents severity
+      required: true
+      type: string
+      enum:
+        - High
+        - Informational
+        - Low
+        - Medium
+    status:
+      title: Incidents status
+      description: Incidents status
+      required: true
+      type: string
+      enum:
+        - Active
+        - Closed
+        - New
+    systemData:
+      description: Azure Resource Manager metadata containing createdBy and modifiedBy
+        information.
+      title: Azure Resource Manager metadata containing createdBy and modifiedBy information.
+      type: SystemData
+    title:
+      description: The title of the incident
+      title: The title of the incident
+      type: string
+    type:
+      description: Azure resource type
+      title: Azure resource type
+      type: string
+  Incident:
+      id:
+        title: Full Incident ID
+        description: Full incident ID
+        type: string
+      name:
+        title: Incident ID
+        description: Incident name - short id
+        type: string
+      type:
+        title: type
+        type: string
+      etag:
+        title: etag
+        type: string
+      properties:
+        title: Incident properties
+        description: Incident properties object.
+        type: IncidentProperties
+
+connection:
+  host:
+    title: Host
+    type: string
+    description: Azure REST API Server
+    required: true
+    default: https://management.azure.com
+  client_id:
+    title: Client ID
+    type: string
+    description: The application ID that the application registration portal assigned
+      to your app
+    required: true
+  client_secret:
+    title: Client Secret
+    type: credential_secret_key
+    description: The application secret that you generated for your app in the app
+      registration portal
+    required: true
+  tenant_id:
+    title: Tenant ID
+    type: string
+    description: This is active directory ID
+    required: true
+  api_version:
+    title: API Version
+    type: string
+    description: The version of the API to use. The current version is 2016-04-30-preview
```

Comment:
Does this ever change? Why would a user want to change this? I'm wondering if we should not allow a user to change this - my gut feeling is that changing this would change the inputs and outputs and break the plugin entirely.